springwolf-plugins/springwolf-cloud-stream-plugin/src/main/java/io/github/springwolf/plugins/cloudstream/asyncapi/scanners/channels/CloudStreamFunctionChannelsScanner.java

```diff
@@ -42,7 +42,7 @@ public class CloudStreamFunctionChannelsScanner implements ChannelsScanner {
 
     private final AsyncApiDocketService asyncApiDocketService;
     private final BeanMethodsScanner beanMethodsScanner;
-    private final ComponentClassScanner componentClassScanner;
+    private final SpringwolfClassScanner classScanner;
```

Comment:
The use of `ComponentClassScanner` and not `SpringwolfClassScanner` is intentional.
`SpringwolfClassScanner` will return classes annotated with `@Component` which is needed here, but also classes returned by bean methods. In the case of spring cloud stream functions, the bean method always has generics, which can't be retrieved (to my knowledge) from the `Class` object.